### DISCUSSION TITLE
Ghost Overtime as a base mechanic

### DISCUSSION THREAD

**Discussion by kassibuss** [NeotokyoRebuild/neo#646] [Ideas]
Since ghost overtime has been a thing for a bit and it sees quite a bit of use, should we add it into the rebuild as a base mechanic? Either as a default in both pub/comp or hide it behind a cvar.

**Comment by brysondev** [comment]
It's hard to see any downsides imo, players see the round taking a bit longer and see a target to kill immediately sounds like more of a positive than a negative. 

"Overtime activated" and everyone spins around chase the funny red dot like a cat.

**Comment by blaberry** [comment]
Agiel posted some interesting commentary, paraphrasing:
"The point of overtime was the opposite. we move regular time into overtime to make rounds where the ghost isn't played shorter. We made ghost OT a thing, but never shortened rounds without ghost play" (iirc rounds were shortened slightly before ghost OT was a thing, but that's beside the point that it wasn't further shortened with the introduction of ghost OT)


I worry that shortening round times further would be bad for comp, where positioning and strategy shouldn't take a backseat to a "we need ghost control right now for the round to not be over" mentality where everything becomes a rush for the ghost (I'm exagreggating a bit here, bear with me)

But, I think shortening pub rounds without ghost play is an interesting idea.

**Reply by kassibuss** [reply]
In regards to comp, I think the base game/pub should be as close to comp as possible. 

I think the current round time in comp* with the OT included is good

**Reply by xedmain** [reply]
imo current comp time of 2:00 facilitates just what you described, but also instead of grabbing the ghost, what's done is just afking at spawn and letting rounds tie out

**Reply by xedmain** [reply]
> In regards to comp, I think the base game/pub should be as close to comp as possible.
> 
> I think the current round time with the OT included is good

pub's 2:30 works out great rn yeah

**Comment by Agiel** [comment]
So the idea of overtime was twofold. The problem I saw in comp at the time was that at the end of a round, when one team was behind they'd often fall back and camp to let the timer run out and get a tie rather than a loss. By adding overtime I wanted to solidify ghost play as the counter to that by giving the other team enough time to hunt the last opponents down and close out the round.

Now, with the extra overtime as a variable to play around with I got the idea of speeding up the game. Basically it lets us get the "boring" rounds over with faster, while having "exciting" rounds stay the same. (Purely subjectively assigning exciting = ghost in play). The standard comp round time then was 2:30, so my idea was to take 45 seconds of that (bomb timer in CS) and assign it to ghost play, leaving 1:45 as the regular round time. Somehow we ended up with 2:00 + 45, which I think is too much, but that's another story...

**Reply by Agiel** [reply]
Speaking of "exciting", the grace time left on the clock when the ghost is dropped plays an important role in the implementation. You may have noticed that overtime starts at 15 seconds rather than at 0, and the timer keeps displaying this decaying grace time, rather than replacing it with the actual remaining overtime. In the spectator overlay I added a glitch effect to the timer during overtime for some extra flavor. Basically the timer is hacked to not run out. 

I think there's value to keeping this somewhere, even if we do end up replacing the main timer with the overtime. Have the decaying grace time off to the side somewhere, to let players know how much time they'll have left if the ghost is dropped.

**Reply by AdamTadeusz** [reply]
I've played many games where a round goes into overtime and its just really difficult to understand what is going on during overtime. Clock says theres x seconds left but it doesnt budge, really awkward to work out how much time is actually left. I would much rather the clock counted down uniformly regardless of whether its normal time or over time

**Reply by AdamTadeusz** [reply]
https://github.com/user-attachments/assets/d58066eb-3572-4e8f-98c2-08fc3ee66bf4

I think something like this with a 5 second grace time from when the ghost is dropped, but ensuring the timer continues normally in the background so grace time doesn't extend overtime time would be easier to understand, with perhaps the grace timer being placed somewhere else

**Reply by Agiel** [reply]
> I've played many games where a round goes into overtime and its just really difficult to understand what is going on during overtime. Clock says theres x seconds left but it doesnt budge, really awkward to work out how much time is actually left. I would much rather the clock counted down uniformly regardless of whether its normal time or over time

This is very much intentional. It's supposed to be overtime and not part of regular round time, and the uncertainty is what brings the extra excitement. How much the round is extended depends on how well you're playing the objective (I know that's not technically true, but this what I wanted to convey to the players, and how most other games do it). In the end though, most players are conservative and expressed the annoyance like you did of not knowing to the second how much time there's left, so I regrettably had to add the timer in the bottom left chat area.

**Comment by Sakhanist** [comment]
No.
Get your comp shit out of pubs.

Tie mechanic existed.

**Reply by AdamTadeusz** [reply]
Yeah I think having over time as an extra feature that has to be manually enabled by the server host is a good idea

**Reply by blaberry** [reply]
I've always insisted on respectful communication in the past, even when it means setting boundaries earlier than I would have preferred. The focus is on completing the reimplementation of nt;re, not dealing with inappropriate behavior. If you are having trouble maintaining a respectful and constructive tone in your feedback, please refrain from participating until you're able to do so.

**Reply by Sakhanist** [reply]
> Yeah I think having over time as an extra feature that has to be manually enabled by the server host is a good idea

At maximum. Feature is overall annoying for maptime. Same dragging out but instead of +2:30 you get +45 +2:30 based on how assholish you want to get.

**Reply by brysondev** [reply]
> At maximum. Feature is overall annoying for maptime. Same dragging out but instead of +2:30 you get +45 +2:30 based on how assholish you want to get.

So the giant red icon on everyone's HUD indicated a possible free kill would apparently dragging out pubs? Can you elaborate on how your argument is even a possibility when nobody even plays ghost in pubs to begin with? If anything this incentivizes more ghost play in pubs by possibly "trolling" the lobby with a giant target on their head.  

"Tie mechanic exists" so constantly tying games is good how? No xp gains because people sat out of a fight all game isn't exactly better for pubs 🤔

**Reply by Agiel** [reply]
> +45 +2:30

Your problem isn't overtime, it's these values. 45 seconds is borderline too much even for comp. It should be lowered to 30. And 2:30 regular time is a drag even without overtime.

**Reply by brysondev** [reply]
Also ftr, having this as a cvar will happen if we want this implemented. I completely agree it's not pre-existing functionality in og, so it should optional.

**Reply by Sakhanist** [reply]
> So the giant red icon on everyone's HUD indicated a possible free kill would apparently dragging out pubs? Can you elaborate on how your argument is even a possibility when nobody even plays ghost in pubs to begin with? If anything this incentivizes more ghost play in pubs by possibly "trolling" the lobby with a giant target on their head.

> "Tie mechanic exists" so constantly tying games is good how? No xp gains because people sat out of a fight all game isn't exactly better for pubs 🤔

It's a team deathmatch on a timer. Ties are not in the game because they are good, they are in the game because no other countable value to indicate advantage to advance the map other than remaining players. Sometimes it's an uneventful 1v1. Yes, I'd rather take 2:30 more in pubs than 2:30 + any amount of 'haha i picked up the ghost I'll drop it the second I see someone'. 

TLDR: I'd take the fixed round timer over the dynamic meme.


> Your problem isn't overtime, it's these values. 45 seconds is borderline too much even for comp. It should be lowered to 30. And 2:30 regular time is a drag even without overtime.

Nah, it's the overtime. Too many rules for a chill pew-pew + occasional CTF.
Since CS analogy has been thrown around the initial discussion of overtimes:
you either get a fun retake or 2-3 vs 12 saving for god knows what.
in sweaty cs it's just a rotation + retake + defuse timer or 40 secs of saving.
there is no flipping sides in NT, there is only tying or proper W-holding.
